src/build.d

```diff
@@ -414,6 +433,33 @@ auto predefinedTargets(string[] targets)
     return newTargets.data;
 }
 
+// An input range for a recursive set of dependencies
+struct DependencyRange
+{
+    private DependencyRef[] next;
+    private bool[DependencyRef] added;
+    this(DependencyRef[] deps) { addDeps(deps); }
+    bool empty() const { return next.length == 0; }
+    auto front() inout { return next[0]; }
+    void popFront()
+    {
+        auto save = next[0];
+        next = next[1 .. $];
+        addDeps(save.deps);
+    }
+    void addDeps(DependencyRef[] deps)
```

Comment:
DDoc comment for the benefit of those using editors that support DDoc features.

Reply:
Actually, I don't see any need for this to be a function, can't you just put it into the constructor?

Reply:
It's called from the constructor and from popFront